src/pypdfium2/_helpers/textpage.py

```diff
@@ -201,45 +160,21 @@ def get_charbox(self, index, loose=False):
             left, bottom, right, top = left.value, bottom.value, right.value, top.value
         
         if not success:
-            raise PdfiumError("Retrieving the char box failed")
+            raise PdfiumError("Retrieving the char box failed.")
         
         return left, bottom, right, top
     
     
-    def get_rectboxes(self, index=0, count=0):
-        """
-        Get the bounding boxes of text rectangles in the requested scope.
-        
-        Yields:
-            Coordinates for left, bottom, right, and top (as :class:`float` values).
-        """
-        n_rects = self.count_rects(index, count)
-        for index in range(n_rects):
```

Comment:
This was very wrong. We can't just do `range(n_rects)` here if `index` is greater than 0. We also need to take into account how many rects there are before `index`. The new function in the test code shows how it should have been:
```python
    if search_result is None:
        return
    c_index, c_count = search_result
    r_index = textpage.count_rects(0, c_index)
    r_count = textpage.count_rects(c_index, c_count)
    rects = [textpage.get_rect(i) for i in range(r_index, r_index+r_count)]
    return rects
```